syft/internal/fileresolver/directory_indexer.go

```diff
@@ -443,11 +450,50 @@ func (r *directoryIndexer) disallowRevisitingVisitor(_, path string, _ os.FileIn
 	return nil
 }
 
-func disallowUnixSystemRuntimePath(base, path string, _ os.FileInfo, _ error) error {
-	// note: we need to consider all paths relative to the base when being filtered, which is how they would appear
-	// when the resolver is being used. Then something like /some/mountpoint/dev with a base of /some/mountpoint
-	// would be considered as /dev when being filtered.
-	if internal.HasAnyOfPrefixes(relativePath(base, path), unixSystemRuntimePrefixes...) {
+type unixSystemMountFinder struct {
+	disallowedMountPaths []string
+}
+
+func newUnixSystemMountFinder() unixSystemMountFinder {
+	infos, err := mountinfo.GetMounts(nil)
+	if err != nil {
+		log.WithFields("error", err).Warnf("unable to get system mounts")
+		return unixSystemMountFinder{}
+	}
+
+	return unixSystemMountFinder{
+		disallowedMountPaths: keepUnixSystemMountPaths(infos),
+	}
+}
+
+func keepUnixSystemMountPaths(infos []*mountinfo.Info) []string {
+	var mountPaths []string
+	for _, info := range infos {
+		if info == nil {
+			continue
+		}
+		// on a unix system you may see:
+		// - tmpfs: a memory-based filesystem (e.g. /tmp, /dev, /sys/firmware)
+		// - devtmpfs: a virtual filesystem that provides access to devices (e.g. /dev)
+		// - shm: a shared memory filesystem (e.g. /dev/shm)
+		// - proc: a virtual filesystem that provides access to process information (e.g. /proc)
+		// - sysfs: a virtual filesystem that provides access to kernel information (e.g. /sys)
+
+		// on a darwin system you may see:
+		// - devfs: a virtual filesystem that provides access to devices (e.g. /dev)
+
+		switch info.FSType {
+		case "tmpfs", "devfs", "devtmpfs", "shm", "proc", "sysfs":
```

Comment:
Things I'm seeing in `df -a` that aren't in here but maybe should be (running in a lima ubuntu instance if that matters):

* udev
* devpts (seems to be under `/dev` so I don't know that we need it.)